### PR TITLE
UI Redesign

### PR DESCRIPTION
**Pull request by jnunemaker**
I've been 100% using [flipper cloud](https://www.flippercloud.io) at this point but when I go back to the OSS UI I miss cloud, so I thought I'd spend some time refreshing the OSS version to work/look a bit more like it. 

## Old Features List

A wall of text.

old-list

## New Features List

Compact and succinct. In one quick look you can tell exactly how the feature is enabled without having to dive into the show view.

new-list

## Old Feature Show

Everything you could possibly want to do all shown to you at once. 

old-show

## New Feature Show

Again, succinct. Add an actor/group or change a % is just a click, but you get a much more simple view now. No longer does the danger zone draw your eye, but it is still there if you need it.

new-show

When fully enabled, all the other gates are hidden to keep the view even more simple and obvious.

new-show-fully-enabled

## Wrap Up

This also all works nicely on mobile from my testing. Open to any thoughts/review and I'll probably merge and cut a new release soon.

### CODE REVIEW

File: lib/flipper/ui/views/layout.erb [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <![endif]-->
   </head>
   <body>
-    <div class="container">
+    <div class="container" style="max-width:600px;">
```
Comment (thetimbanks) [inline lib/flipper/ui/views/layout.erb:19]: Curious about the decision to go with the max width here. I assume the new layout just looks a lot better at a skinnier width compared to the old layout?

File: examples/ui/basic.ru [hunk 2]
```diff
@@ -35,6 +35,10 @@ adapter = Flipper::Adapters::PStore.new
 flipper = Flipper.new(adapter, instrumenter: ActiveSupport::Notifications)
 
 Flipper::UI.configure do |config|
+  # config.banner_text = 'Production Environment'
+  # config.banner_class = 'danger'
```
[the remark this answers is not in the document]
Comment (jnunemaker) [reply examples/ui/basic.ru:39]: I was testing and changing them back and forth. Left them so people could see full example of settings. Do you think I should remove them or are they ok commented out?

File: lib/flipper/ui/public/js/application.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 (function() {
   $(function() {
-    return $('[data-toggle="tooltip"]').tooltip();
+    return $(document).on('click', '.js-toggle-trigger', function() {
+      var $container;
+      $container = $(this).closest('.js-toggle-container');
```
[the remark this answers is not in the document]
Comment (jnunemaker) [reply lib/flipper/ui/public/js/application.js:5]: Yeah good point. This is auto generated from the coffeescript file so I don't have control over that. I should just switch to plain old js like the cool kids. Flipper UI is old enough that coffeescript was in style.

File: lib/flipper/ui/views/layout.erb (hunk 1 above)
Comment (jnunemaker) [reply lib/flipper/ui/views/layout.erb:19]: Yep, I thought it looked much better shrunk up. It was just really long empty space and I couldn't come up with anything very valuable to fill it. It felt like a terrible way to do it so I'm open to a more standard way.

**Comment by jnunemaker** [comment]
@thetimbanks I **really** appreciate you taking the time to look this over and so quickly. Thanks! I'll tweak up anything you suggest and we can go from there.

File: examples/ui/basic.ru (hunk 2 above)
[the remark this answers is not in the document]
Comment (thetimbanks) [reply examples/ui/basic.ru:39]: This is probably ok to leave in then.

File: lib/flipper/ui/views/layout.erb (hunk 1 above)
Comment (thetimbanks) [reply lib/flipper/ui/views/layout.erb:19]: I don't think there is anything wrong with setting a max width for the app if it looks good on all screens at that width. I generally try to stay away from inline styles. You could add another class to that `div` or set the `max-width` of the `container` class to `600px`. I don't know if that `container` class is used elsewhere though.

File: lib/flipper/ui/public/js/application.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (thetimbanks) [reply lib/flipper/ui/public/js/application.js:5]: Ha, I did cringe a little at the coffescript but I knew that was the default in Rails for awhile.

File: lib/flipper/ui/views/layout.erb (hunk 1 above)
Comment (jnunemaker) [reply lib/flipper/ui/views/layout.erb]: Switched to a class in 011e63d

File: lib/flipper/ui/public/js/application.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jnunemaker) [reply lib/flipper/ui/public/js/application.js]: The nudge was all I needed. Removed coffeescript and sass in b2fa5e6. It was bothering me already anyway.

**Pull request merged**